ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -146,24 +146,20 @@ StatusDialog {
                 }
 
                 currencyStore: root.swapAdaptor.currencyStore
-                flatNetworksModel: root.swapAdaptor.filteredFlatNetworksModel
-                processedAssetsModel: root.swapAdaptor.processedAssetsModel
+                flatNetworksModel: root.swapAdaptor.swapStore.flatNetworks
+                processedAssetsModel: root.swapAdaptor.walletAssetsStore.groupedAccountAssetsModel
 
                 tokenKey: root.swapInputParamsForm.fromTokensKey
-                tokenAmount: {
-                    // Only update if there is different in amount displayed
-                    if (root.swapInputParamsForm.fromTokenAmount !==
-                            SQUtils.AmountsArithmetic.fromString(value).toLocaleString(locale, 'f', -128)){
-                        return root.swapInputParamsForm.fromTokenAmount
-                    }
-                    return payPanel.tokenAmount
-                }
+
+                selectedNetworkChainId: root.swapInputParamsForm.selectedNetworkChainId
+                selectedAccountAddress: root.swapInputParamsForm.selectedAccountAddress
+                nonInteractiveTokensKey: receivePanel.selectedHoldingId
 
                 swapSide: SwapInputPanel.SwapSide.Pay
                 swapExchangeButtonWidth: swapButton.width
 
                 onSelectedHoldingIdChanged: root.swapInputParamsForm.fromTokensKey = selectedHoldingId
-                onValueChanged: root.swapInputParamsForm.fromTokenAmount = value.toLocaleString(locale, 'f', -128)
+                onRawValueChanged: Qt.callLater(() => root.swapInputParamsForm.fromTokenAmount = rawValue)
```

Comment:
by rawValue do we mean value in wei?

Reply:
I am not sure why we changed this?

Reply:
its is wei as we discussed, then in that case we need to convert it before we assign it back to popup on line 69, either always amount in wei or not.